src/status_im2/subs/contact.cljs

```diff
@@ -32,28 +32,41 @@
    (get multiaccount :profile-pictures-visibility)))
 
 (defn- replace-contact-image-uri
-  [contact port identity]
-  (let [theme          (theme/get-theme)
-        contact-images (:images contact)
-        contact-images (reduce (fn [acc image]
-                                 (let [image-name (:type image)
-                                       ; We pass the clock so that we reload the image if the image is
-                                       ; updated
-                                       clock      (:clock image)
-                                       uri        (image-server/get-contact-image-uri port
-                                                                                      identity
-                                                                                      image-name
-                                                                                      clock
-                                                                                      theme)]
-                                   (assoc-in acc [(keyword image-name) :uri] uri)))
-                               contact-images
-                               (vals contact-images))]
-    (assoc contact :images contact-images)))
+  [contact port public-key font-file]
+  (let [theme (theme/get-theme)
+        {:keys [images ens-name]} contact
+        images
+        (reduce (fn [acc image]
+                  (let [image-name (:type image)
+                        ; We pass the clock so that we reload the image if the image is
+                        ; updated
+                        clock      (:clock image)
+                        uri        (image-server/get-contact-image-uri-fn
+                                    {:port           port
+                                     :public-key     public-key
+                                     :image-name     image-name
+                                     :clock          clock
+                                     :theme          theme
+                                     :override-ring? (if ens-name false nil)})]
+                    (assoc-in acc [(keyword image-name) :fn] uri)))
+                images
+                (vals images))
+
+        images (if (seq images)
+                 images
+                 [{:fn (image-server/get-initials-avatar-uri-fn
+                        {:port           port
+                         :public-key     public-key
+                         :override-ring? (if ens-name false nil)
+                         :theme          theme
+                         :font-file      font-file})}])]
+
+    (assoc contact :images images)))
```

Comment:
Another comment, is we are shadowing the `images` var, which is not desirable in Clojure, if we shadow a var it's like "mutating" the var, since it has a different value across the function.

as a first approach to refactor this function I got this, take it or discard it, just a suggestion: :smiley: 

```clojure
(defn- replace-contact-image-uri
  [{:keys [images ens-name] :as contact} port public-key font-file]
  (let [theme           (theme/get-theme)
        images-with-uri (some->> images
                          (vals)
                          (reduce (fn [acc {image-name :type clock :clock}]
                                    (let [uri (image-server/get-contact-image-uri-fn
                                               {:port           port
                                                :public-key     public-key
                                                :image-name     image-name
                                                ; We pass the clock so that we reload the
                                                ; image if the image is updated
                                                :clock          clock
                                                :theme          theme
                                                :override-ring? (when ens-name false)})]
                                      (assoc-in acc [(keyword image-name) :fn] uri)))
                                  images))
        contact-images  (or images-with-uri
                            [{:fn (image-server/get-initials-avatar-uri-fn
                                   {:port           port
                                    :public-key     public-key
                                    :theme          theme
                                    :override-ring? (when ens-name false)
                                    :font-file      font-file})}])]
    (assoc contact :images contact-images)))
```

Reply:
> we are changing the structure of the :images key inside a contact

Fixed. It's not what I intend to do, changed it back to map
The `:images` in multiaccount is a vector and I thought it must be a vector in contact as well and didn't check.